### PR TITLE
Fix daemonset dependency panic when POD_NAME is unset

### LINKED ISSUE #15: Daemonset dependency not working.
When running kubernetes-entrypoint in init-container daemoset dependency is returning nil *Daemonset pointer error.

### PR DESCRIPTION
**Pull request by mzylowski**
Closes #15

### CODE REVIEW

File: dependencies/daemonset/daemonset.go [hunk 1]
```diff
@@ -23,15 +23,16 @@ func init() {
...
 			}
-			entry.Register(daemonset)
 		}
 	}
 }
 
 func NewDaemonset(name string) (*Daemonset, error) {
 	if os.Getenv("POD_NAME") == "" {
-		return nil, fmt.Errorf("Env POD_NAME not set")
+		return nil, fmt.Errorf("Env POD_NAME not set. Daemonset %s will be ignored!", name)
```
Comment (PiotrProkop) [inline dependencies/daemonset/daemonset.go]: /Daemonset/Daemonset dependency/

File: dependencies/daemonset/daemonset.go [hunk 2]
```diff
@@ -23,15 +23,16 @@ func init() {
 			daemonset, err := NewDaemonset(dep)
 			if err != nil {
 				logger.Error.Printf("Cannot initialize daemonset: %v", err)
+			} else {
+				entry.Register(daemonset)
 			}
-			entry.Register(daemonset)
```
Comment (PiotrProkop) [inline dependencies/daemonset/daemonset.go]: instead of doing else statement just put continue after printing error.

**Comment by mzylowski** [comment]
@PiotrProkop PTAL again

**Comment by PiotrProkop** [comment]
LGTM

**Pull request merged**